os/fs/vfs/fs_read.c

```diff
@@ -111,23 +117,16 @@ ssize_t file_read(FAR struct file *filep, FAR void *buf, size_t nbytes)
 	 * method?
 	 */
 
-	else if (inode && inode->u.i_ops && inode->u.i_ops->read) {
+	else if (inode != NULL && inode->u.i_ops && inode->u.i_ops->read) {
 		/* Yes.. then let it perform the read.  NOTE that for the case of the
 		 * mountpoint, we depend on the read methods being identical in
 		 * signature and position in the operations vtable.
 		 */
 
-		ret = (int)inode->u.i_ops->read(filep, (char *)buf, (size_t)nbytes);
-	}
-
-	/* If an error occurred, set errno and return -1 (ERROR) */
-
-	if (ret < 0) {
```

Comment:
uh oh, you removed code here, then file_seek and another function also need to be checked.